### ISSUE TITLE
.finally() method error on Promise returned by createUserWithEmailAndPassword()

### ISSUE THREAD

**Issue by darkylmnx** [firebase/firebase-js-sdk#543]
### [REQUIRED] Describe your environment

  * Operating System version:  Windows 10 / Chrome 64
  * Firebase SDK version:  4.10.1
  * Firebase Product: auth

### [REQUIRED] Describe the problem
In an app i'm creating, using vuejs and bluebird to polyfill Promise on the browser, i did the following : 
```
firebase.auth()
    .createUserWithEmailAndPassword('aa@aa.aa', 'aa@aa.aa')
    .then( ... )
    .catch( ... )
    .finally( ... )
```

I get an error saying that **finally** isn't a method, though `window.Promise.prototype.finally` exists on my browser.

From your doc it's written : 

> firebase.Promise is the same as the native Promise implementation when available in the current environment, otherwise it is a compatible implementation of the Promise/A+ spec

I have a native (Polyfilled Promise version) but it seems firebase uses it's own version, even when i do : 
`firebase.Promise = require('bluebird')` OR `firebase.Promise = window.Promise` it doesn't seem to work.

There's an issue here as it doesn't work as expected based on the doc.

#### Relevant Code:
https://jsfiddle.net/dyo983az/5/ & https://jsfiddle.net/dyo983az/8/

**Comment by bojeil-google** [comment]
This applies for all promises returned by Auth async APIs. Auth uses `goog.Promise` for our implementation to support old browsers.  It has its own version of `finally` but is not exposed. For now, you can easily just add your onCompletion in the last resolve and reject.

**Comment by darkylmnx** [comment]
@bojeil-google well that's what i did but it's not DRY, why is finally not exposed ? It's part of the Promise spec and it's a must have for developers to do tasks when the promise is over especially when there's a lot of chaining over/under the hood

**Comment by bojeil-google** [comment]
This was fixed in version 4.12.0

**Comment by darkylmnx** [comment]
Okay I'll give it a try.

Le lun. 26 mars 2018 à 23:28, bojeil-google <notifications@github.com> a
écrit :
